### PR TITLE
Add pg port to specified pgaddr without port

### PR DESCRIPTION
**Pull request by nak3**
[Review on Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/3911)

### CODE REVIEW

File: sql/pgwire/server.go [hunk 1]
```diff
@@ -36,6 +37,8 @@ const (
 	versionSSL = 80877103
 )
 
+const pgPort = 15432
```
Comment (petermattis) [inline sql/pgwire/server.go]: Seems a bit strange that we're duplicating the port here and in `server/context.go`. Can we move this logic into `server` near the usage of `context.PGAddr`? And why is this necessary for `context.PGAddr` but not `context.Addr`? Or maybe it is necessary for `context.Addr` but I'm just not seeing where it is done.

**Comment by nak3** [comment]
Thank you @petermattis How about this one?

File: cli/cli.go [hunk 2]
```diff
@@ -50,9 +50,6 @@ Output build version information.
 
 var cockroachCmd = &cobra.Command{
 	Use: "cockroach",
-	PersistentPreRun: func(cmd *cobra.Command, args []string) {
-		context.Addr = util.EnsureHostPort(context.Addr)
-	},
```
Comment (petermattis) [inline cli/cli.go]: Ah, this is where we ensure a port on `context.Addr`. I'm curious who you didn't just add a line here to ensure a port on `context.PGAddr`. You'd have to change `EnsureHostPort` to take a port to append if one isn't present. As it is, there is a lot of duplicated functionality between `EnsureHostPort` and your new `portPadding` function.

Comment (mberhault) [inline cli/cli.go]: I'm separating addr and port in
https://github.com/cockroachdb/cockroach/pull/3757
Part of that change involves proper defaults, and always concatenating host
and port.

**Comment by nak3** [comment]
Thank you again @petermattis. I just hesitated to change `EnsureHostPort` to take a port to append. I updated `EnsureHostPort` and used it now.

@mberhault Although I updated here after @petermattis's review , if your PR will fix this, I'm OK with taking your patch.

**Comment by tamird** [comment]
@mberhault can you take a look at this now that #3757 has landed?

**Comment by tamird** [comment]
LGTM
